test/models/compute_resource_test.rb

```diff
@@ -224,6 +224,11 @@ def setup
     assert_equal host, as_admin { cr.send(:associate_by, 'mac', '00:22:33:44:55:1a') }
   end
 
+  test 'valid if url is blank and provider is not eligible to have URL' do
+    cr = FactoryGirl.build(:vmware_cr)
```

Comment:
Could you check that instead the `server` attribute is required and a error message is added for `server`. Maybe `shoulda-matchers` help to reduce the boilerplate.